src/tpm.rs

```diff
@@ -718,62 +719,173 @@ impl Tpm for TpmTss {
             }
         };
 
-        self.execute_with_temporary_object_context(mk_key_context, |hsm_ctx, mk_key_handle| {
-            hsm_ctx
-                .tpm_ctx
-                .create(mk_key_handle.into(), key_pub, None, None, None, None)
-                .map(
-                    |CreateKeyResult {
-                         out_private: private,
-                         out_public: public,
-                         creation_data: _,
-                         creation_hash: _,
-                         creation_ticket: _,
-                     }| {
-                        match algorithm {
-                            KeyAlgorithm::Ecdsa256 => LoadableIdentityKey::TpmEcdsa256V1 {
-                                private,
-                                public,
-                                x509: None,
-                            },
-                            KeyAlgorithm::Rsa2048 => LoadableIdentityKey::TpmRsa2048V1 {
-                                private,
-                                public,
-                                x509: None,
-                            },
-                        }
-                    },
-                )
-                .map_err(|tpm_err| {
-                    error!(?tpm_err);
-                    TpmError::TpmIdentityKeyCreate
-                })
-        })
+        if let Some(auth_value) = auth_value {
+            let tpm_auth_value = match auth_value {
+                AuthValue::Key256Bit { auth_key } => Auth::from_bytes(auth_key.as_ref()),
+            }
+            .map_err(|tpm_err| {
+                error!(?tpm_err);
+                TpmError::TpmAuthValueInvalid
+            })?;
+
+            self.execute_with_temporary_object_context(
+                mk_key_context,
+                |hsm_ctx, primary_key_handle| {
+                    let (private, public) = hsm_ctx
+                        .tpm_ctx
+                        .create(
+                            primary_key_handle.into(),
+                            key_pub,
+                            Some(tpm_auth_value.clone()),
+                            None,
+                            None,
+                            None,
+                        )
+                        .map(
+                            |CreateKeyResult {
+                                 out_private: private,
+                                 out_public: public,
+                                 creation_data: _,
+                                 creation_hash: _,
+                                 creation_ticket: _,
+                             }| { (private, public) },
+                        )
+                        .map_err(|tpm_err| {
+                            error!(?tpm_err);
+                            TpmError::TpmIdentityKeyCreate
+                        })?;
+
+                    let storage_key_pub = Self::create_storage_key_public()?;
```

Comment:
I suspect this `storage_key_pub` is causing the error, but I don't know the correct parameters to build one of these. Reusing (or recreating) `key_pub` from above causes the same error still.